File: CSharpCodeBase/utils/bhlib/sequence.lua
local Class = require 'utils.hump.class'

local Sequence = Class({init = function(self, behaviours)
      self._behaviours = behaviours
      self.returnCode = B_FAILURE
end})

function Sequence:Behave(creature)
  
  anyRunning = false

  for i = 1, #self._behaviours do
    code = self._behaviours[i]:Behave(creature)
    if code == B_FAILURE then
      self.returnCode = code
      return self.returnCode
    elseif code == B_SUCCESS then
      -- continue
    elseif code == B_RUNNING then
      anyRunning = true
      -- continue
    end
  end
  self.returnCode = anyRunning and B_RUNNING or B_SUCCESS
  return self.returnCode
end

return Sequence

File: CSharpCodeBase/utils/bhlib/sequence.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class sequence {
 public void init(self, behaviours);
       this._behaviours = behaviours;
       this.returnCode = B_FAILURE;
 }})
 public void Behave(creature){
   
   anyRunning = false;
   for(i = 1, #this._behaviours ){
     code = this._behaviours[i]:Behave(creature);
     if(code == B_FAILURE  ){ 
       this.returnCode = code;
       return this.returnCode;
     }else{if code == B_SUCCESS  ){ 
       // continue
     }else{if code == B_RUNNING  ){ 
       anyRunning = true;
       // continue
     }
   }
   this.returnCode = anyRunning  &&  B_RUNNING  ||  B_SUCCESS;
   return this.returnCode;
 }
 }}